torchdata/datapipes/iter/load/iopath.py

```diff
@@ -1,73 +1,108 @@
 # Copyright (c) Facebook, Inc. and its affiliates.
 import os
 
+from typing import Iterator, List, Tuple, Union
+
+from torch.utils.data.datapipes.utils.common import match_masks
+
 from torchdata.datapipes import functional_datapipe
 from torchdata.datapipes.iter import IterDataPipe
 from torchdata.datapipes.utils import StreamWrapper
-from typing import Iterator, Tuple
+
+try:
+    import iopath
+
+except ImportError:
+    iopath = None
+
+
+def _create_default_pathmanager():
+    from iopath.common.file_io import HTTPURLHandler, OneDrivePathHandler, PathManager
+    pathmgr = PathManager()
+    pathmgr.register_handler(HTTPURLHandler(), allow_override=True)
+    pathmgr.register_handler(OneDrivePathHandler(), allow_override=True)
+    # S3PathHandler is not included in 0.1.8
+    try:
+        from iopath.common.s3 import S3PathHandler
+
+        pathmgr.register_handler(S3PathHandler(), allow_override=True)
+    except ImportError:
+        pass
+    return pathmgr
 
 
 class IoPathFileListerIterDataPipe(IterDataPipe[str]):
     r""":class:`IoPathFileListerIterDataPipe`.
 
-    Iterable DataPipe to list the contents of the directory at the provided
-    `root` URI.
-    pathnames. This yields the full URI for each file within the directory.
+    Iterable DataPipe to list the contents of the directory at the provided `root` pathname or url,
+    and yields the full pathname or url for each file within the directory.
+
     Args:
-        root: The base URI directory to list files from
+        root: The root local filepath or url directory to list files from
+        masks: Unix style filter string or string list for filtering file name(s)
+
+    Note:
+        This IterDataPipe currently supports local file path, normal HTTP url and OneDrive url.
+        S3 url is supported only with `iopath`>=0.1.9.
     """
 
-    def __init__(self, *, root: str) -> None:
-        try:
-            from iopath.common.file_io import g_pathmgr
-        except ImportError:
+    def __init__(
+        self,
+        root: str,
+        masks: Union[str, List[str]] = "",
+    ) -> None:
+        if iopath is None:
             raise ModuleNotFoundError(
                 "Package `iopath` is required to be installed to use this "
-                "datapipe. Please use `pip install iopath` or `conda install "
-                "iopath`"
-                "to install the package"
+                "datapipe. Please use `pip install iopath` to install the package"
             )
 
         self.root: str = root
-        self.pathmgr = g_pathmgr
+        self.pathmgr = _create_default_pathmanager()
+        self.masks = masks
+
+    def register_handler(self, handler, allow_override=False):
```

Comment:
Do the existing test cases cover these change? Or do we plan to have a separate PR to write a complete set of test for these DataPipes?

Reply:
It requires to have remote data sources. Can't really any test for now.

Reply:
We only have test for local files.